From 671b27877b22b0e9ba3dd2bdaa0a3909c501161f Mon Sep 17 00:00:00 2001
From: Emmanuel Pescosta <emmanuelpescosta099@gmail.com>
Date: Thu, 3 Apr 2014 02:03:14 +0200
Subject: [PATCH 2/2] Fix crash on startup caused by Dropbox plugin

Create the item state socket on "begin retrieval" and delete
it again in "end retrieval".

BUG: 332481
FIXED-IN: 4.13.0
REVIEW: 117344
---
 dropbox/fileviewdropboxplugin.cpp | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/dropbox/fileviewdropboxplugin.cpp b/dropbox/fileviewdropboxplugin.cpp
index 0026d59..3496ce1 100644
--- a/dropbox/fileviewdropboxplugin.cpp
+++ b/dropbox/fileviewdropboxplugin.cpp
@@ -44,17 +44,11 @@ public:
         contextFilePaths(),
         controlSocketPath(),
         controlSocket(new QLocalSocket(parent)),
-        itemStateSocket(new QLocalSocket),
         databaseFileWatcher(new QFileSystemWatcher(parent)),
         contextActions(new KActionCollection(parent))
     {
     }
 
-    ~Private()
-    {
-        delete itemStateSocket;
-    }
-
     QStringList contextFilePaths;
     QString controlSocketPath;
     QPointer<QLocalSocket> controlSocket;
@@ -106,6 +100,8 @@ bool FileViewDropboxPlugin::beginRetrieval(const QString& directory)
     qRegisterMetaType<QAbstractSocket::SocketError>("QAbstractSocket::SocketError");
     qRegisterMetaType<QAbstractSocket::SocketState>("QAbstractSocket::SocketState");
 
+    d->itemStateSocket = new QLocalSocket;
+
     return connectWithDropbox(d->itemStateSocket, LongTimeout);
 }
 
@@ -123,6 +119,7 @@ KVersionControlPlugin2::ItemVersion FileViewDropboxPlugin::itemVersion(const KFi
 
 void FileViewDropboxPlugin::endRetrieval()
 {
+    delete d->itemStateSocket;
 }
 
 QList<QAction*> FileViewDropboxPlugin::actions(const KFileItemList& items) const
-- 
1.9.1

